core/templates/components/common-layout-directives/navigation-bars/top-navigation-bar.component.spec.ts

```diff
@@ -0,0 +1,452 @@
+    expect(component.labelForClearingFocus).toBe(undefined);
+    expect(component.logoutUrl).toBe(undefined);
+    expect(component.userMenuIsShown).toBe(undefined);
+    expect(component.inClassroomPage).toBe(undefined);
+    expect(component.windowIsNarrow).toBe(false);
+    expect(component.navElementsVisibilityStatus).toEqual({});
+
+    component.ngOnInit();
+    tick();
+
+    expect(component.currentUrl).toBe('learn');
+    expect(component.labelForClearingFocus)
+      .toBe(AppConstants.LABEL_FOR_CLEARING_FOCUS);
+    expect(component.logoutUrl).toBe(AppConstants.LOGOUT_URL);
+    expect(component.userMenuIsShown).toBe(true);
+    expect(component.inClassroomPage).toBe(true);
+    expect(component.windowIsNarrow).toBe(true);
+    expect(component.navElementsVisibilityStatus).toEqual({
+      I18N_TOPNAV_DONATE: true,
+      I18N_TOPNAV_CLASSROOM: true,
+      I18N_TOPNAV_ABOUT: true,
+      I18N_CREATE_EXPLORATION_CREATE: true,
+      I18N_TOPNAV_LIBRARY: true
+    });
+
+    component.ngOnDestroy();
+  }));
+
+  it('should get user info on initialization', fakeAsync(() => {
+    spyOn(cbas, 'fetchClassroomPromosAreEnabledStatusAsync')
+      .and.resolveTo(true);
+    spyOn(wds, 'isWindowNarrow').and.returnValue(true);
+    spyOn(wds, 'getWidth').and.returnValue(700);
+
+    expect(component.isModerator).toBe(undefined);
+    expect(component.isAdmin).toBe(undefined);
+    expect(component.isTopicManager).toBe(undefined);
+    expect(component.isSuperAdmin).toBe(undefined);
+    expect(component.userIsLoggedIn).toBe(undefined);
+    expect(component.username).toBe(undefined);
+    expect(component.profilePageUrl).toBe(undefined);
+
+    component.ngOnInit();
+    tick();
+
+    expect(component.isModerator).toBe(true);
+    expect(component.isAdmin).toBe(false);
+    expect(component.isTopicManager).toBe(false);
+    expect(component.isSuperAdmin).toBe(false);
+    expect(component.userIsLoggedIn).toBe(true);
+    expect(component.username).toBe('username1');
+    expect(component.profilePageUrl).toBe('/profile/username1');
+
+    component.ngOnDestroy();
+  }));
+
+  it('should truncate navbar after search bar is loaded', (done) => {
+    spyOn(component, 'truncateNavbar');
+
+    component.ngOnInit();
+    mockOnSearchBarLoadedEventEmitter.emit();
+
+    setTimeout(() => {
+      expect(component.truncateNavbar).toHaveBeenCalled();
+      done();
+    }, 100);
+
+    component.ngOnDestroy();
+  });
+
+  it('should try displaying the hidden navbar elements if resized' +
+    ' window is larger', () => {
+    let donateElement = 'I18N_TOPNAV_DONATE';
+    spyOn(wds, 'getWidth').and.returnValue(700);
+    component.ngOnInit();
+
+    component.currentWindowWidth = 600;
+    component.navElementsVisibilityStatus[donateElement] = false;
+
+    windowRef.nativeWindow.dispatchEvent(new Event('resize'));
+
+    component.navElementsVisibilityStatus[donateElement] = true;
+
+    component.ngOnDestroy();
+  });
```

Comment:
There aren't any expect statements here. What are we testing?

Reply:
Done